lib/ProductOpener/APIProductServices.pm

```diff
@@ -107,69 +107,99 @@ my %service_functions = (
 	extend_ingredients => \&ProductOpener::Ingredients::extend_ingredients_service,
 	estimate_ingredients_percent => \&ProductOpener::Ingredients::estimate_ingredients_percent_service,
 	analyze_ingredients => \&ProductOpener::Ingredients::analyze_ingredients_service,
+	check_quality => \&check_quality_service,
```

Comment:
It would be best to put the check_quality_service function in the DataQuality.pm module like the other services

In fact we could change the existing check_quality function so that it behaves like a service (you can look at other service functions: it expects a product_ref as a first argument, and a reference to a hash of updated fields as the second argument) , and rename it to check_quality_service().